stjude/proteinpaint
# use esbuild for command line tests of client specs

## Description

This migrates client unit and integration tests to use `esbuild` instead of `webpack`. This brings the dev bundling closer to  prod bundling, since both `esbuild` and `rollup` output standard `esm` bundles, whereas `webpack` output non-standard module format.  (TODO: Work on using esbuild in a separate branch).

Note that this branch mostly affects spec tests and package scripts, there is almost no CI workflow code changes, so the migration risk is low since the same spec tests are still all passing.

NOTE: 
Stop `npm run dev` before testing, then `run npm run reset` from sjpp dir.

To test:
- In the Actions tab, verify passing unit and integration test CIs, as related to this PR branch.
- Use `testrun.html` as usual, this will use the bundle as generated by `client/dev.sh`
- Command-line testing from the `client` dir:
```sh
npm run test:unit
npm run test:integration

# NEW!: supply a custom spec name glob pattern
./test.sh *tvs.*.spec.*
```


## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
